src/main/java/software/amazon/encryption/s3/S3EncryptionClient.java

```diff
@@ -1139,6 +1139,11 @@ public S3EncryptionClient build() {
                             .secureRandom(_secureRandom)
                             .build();
                 } else if (_kmsKeyId != null) {
+                    try {
+                        Class.forName("software.amazon.awssdk.services.kms.KmsClient");
+                    } catch (ClassNotFoundException e) {
+                        throw new RuntimeException("software.amazon.awssdk:kms is required to set up with KMS key", e);
+                    }
                     KmsClient kmsClient = KmsClient.builder()
                             .credentialsProvider(_awsCredentialsProvider)
                             .region(_region)
```

Comment:
I do not think this check is sufficient to prevent the exception.
The KMS Client is an import of this class.

I think the Java Compiled Byte Code will require KMS,
even if a KMS key is not passed,
because the Java class has references to the objects.

But I could be wrong; 
Maybe the Java compiler is fancier than I thought,
and masks the KMS objects unless this if statement is encountered?

@hieuwu did you run a test that proved that this mitigates your issue?

Because the current testing of the S3EC will not cover this.

Reply:
@texastony I got the same issue when run the app. As kms is dependency of S3EC, I would expect that it should fail at Compile Time. However the build is still successful, the exception is thrown only we start running the app or test.

<img width="750" alt="Screenshot 2024-11-01 at 07 12 42" src="https://github.com/user-attachments/assets/f3132a15-71d4-4d2c-b5f9-f5bc4da3ba00">

I have removed the check. Seems making `software.amazon.awssdk:kms` not optional is the way, just make it required as `software.amazon.awssdk:s3` .

Reply:
I suspect this is the correct fix. 
I'll approve CI run.

Reply:
Great, thank you Tony